security/oc_tls.c

```diff
@@ -69,13 +69,50 @@
 #include <mbedtls/debug.h>
 #include <mbedtls/error.h>
 #include <mbedtls/platform.h>
-#endif /* OC_DEBUG */
+#elif OC_DBG_IS_ENABLED || OC_ERR_IS_ENABLED || OC_WRN_IS_ENABLED
+static void
+mbedtls_strerror(int ret, char *buf, size_t buflen)
+{
+  snprintf(buf, buflen, "MBEDTLS_ERR(%d)", ret);
+  return buf;
+}
+#endif /* OC_DBG_IS_ENABLED || OC_ERR_IS_ENABLED || OC_WRN_IS_ENABLED */
 
 #include <stdarg.h>
 #include <stdint.h>
 #include <string.h>
 #include <inttypes.h>
 
+#if OC_DBG_IS_ENABLED || OC_ERR_IS_ENABLED
+#if OC_ERR_IS_ENABLED
+#define MBEDTLS_ERR(mbedtls_func_name, mbedtls_err)                            \
```

Comment:
Are preprocessor functions instead of normal ones necessary?

Reply:
For log line in proper places - yes

Reply:
Yeah, line will be different, though you have the `mbedtls_func_name` label which can be used to uniquely identify the log.